bgbb/wrappers.py

```diff
@@ -39,16 +39,19 @@ def to_abgd_od(params) -> Dict[str, float]:
     return OrderedDict(zip(abgd_names, params))
 
 
+def module_parameter_list(params, mod):
+    return params or list(mod.params_.values())
+
+
 class Rfn:
```

Comment:
This module is interesting, but some of the metaprogramming makes it hard to read. This is mostly monkey-patching dataframe support into the underlying `BetaGeoBetaBinomFitter` class?

Reply:
Right. In the reference implementation the methods use various combinations of explicit arguments vs accessing object attributes of both R/F/N data and fitted model params. This `Rfn` class just tries to wrap them all so that you could pass the dataframe & fitted params as arguments to any of the methods.

It's not even needed for production, but I found it useful for interactive use.